### ISSUE TITLE
loss_kd reduces to zero

### ISSUE THREAD

**Issue by Turlan** [huawei-noah/DAFL#6]
Hi, thanks for releasing the code! I simply ran the training code on mnist.
![image](https://user-images.githubusercontent.com/12882509/66738852-05be9700-eea2-11e9-9215-89f7cac84f45.png)
When training the student network, after a certain training steps, loss_kd reduces to zero. The test accuracy drops by ~1% at the same time. I checked the generated images, it seems that the generator falls into a bad local minima. All the generated images are same.
![image](https://user-images.githubusercontent.com/12882509/66739106-9301eb80-eea2-11e9-9dc5-8c06f2ad52a1.png)
In this case, the generator is useless for learning the student network. Is this a common phenomenon or a bug?

**Comment by HantingChen** [comment]
It's an accidental phenomena, which might result from the instability of GAN training. You can re-training from start or from the checkpoint before it to get better results. 

BTW, this phenomenon only occurs on MNIST dataset. The training is stable in CIFAR-10 and CIFAR-100 dataset.
